ukrainskyi-vitalii/nodejs-aws-shop-react-backend
# Task 7 (Authorization)

1. [Task](https://github.com/rolling-scopes-school/aws/blob/main/aws-developer/07_authorization/task.md)
2. What's done:
- [x] Task 7.1
- [x] Task 7.2
- [x] Task 7.3
- [x] Task 7.4
- [x] Additional (optional) tasks
3. Self-evaluation: 100/100
4. Done / Deadline: 14.07.2024/15.07.2024

Links:
1. [Auth repo](https://github.com/ukrainskyi-vitalii/nodejs-aws-shop-react-backend/tree/task-7/authorization-service)
2. [Client application](https://d2ov4hqhtnmyzx.cloudfront.net/)
3. [FE pull request ](https://github.com/ukrainskyi-vitalii/nodejs-aws-shop-react-frontend/pull/3)
4. [Import API](https://93lnx6q689.execute-api.eu-west-1.amazonaws.com/prod)

Evaluation criteria (70 points for covering all criteria)
- [x] authorization-service is added to the repo, has correct basicAuthorizer lambda and correct AWS CDK Stack
- [x] Import Service AWS CDK Stack has authorizer configuration for the importProductsFile lambda. Request to the importProductsFile lambda should work only with correct authorization_token being decoded and checked by basicAuthorizer lambda. Response should be in 403 HTTP status if access is denied for this user (invalid authorization_token) and in 401 HTTP status if Authorization header is not provided.
- [x] client application is updated to send "Authorization: Basic authorization_token" header on import. Client should get authorization_token value from browser [localStorage](https://developer.mozilla.org/ru/docs/Web/API/Window/localStorage)

**How to check:**

You can add an auth token to the local storage by the browser dev tool.
Vaild token - dWtyYWluc2t5aV92aXRhbGlpPVRFU1RfUEFTU1dPUkQ=
Invalid token - dWtyYWluc2t5aV92aXRhbGlpPVRFU1Q=

1. Try to add a new product without an auth token - you will see an error 401:
![image](https://github.com/user-attachments/assets/f6bdd581-69dd-4618-bed3-3b3bb5978153)
2. Try to add a new product with an invalid auth token - you will see an error 403:
![image](https://github.com/user-attachments/assets/268e56e0-93f8-4cd3-828c-757318ed8fe7)
3. Try to add a valid auth token and the product from the file will be added: 


Additional (optional) tasks:
+30 - Client application should display alerts for the responses in 401 and 403 HTTP statuses. This behavior should be added to the nodejs-aws-fe-main/src/index.tsx file.

You can see it in the [ClientFront Application](https://d2ov4hqhtnmyzx.cloudfront.net/) and in the [FE pull request](https://github.com/ukrainskyi-vitalii/nodejs-aws-shop-react-frontend/pull/3)
